### DISCUSSION TITLE
HeatWave V4 Tutorial fails to create new project

### DISCUSSION THREAD

**Discussion by ColinWPA** [wixtoolset/issues#8103] [Questions] [answered]
I have been successfully creating Wix V4 projects without issue until now, but I had to leave working on this to attend to other projects. On return to Wix, my projects would no longer load, giving an error about mission files or methods. I my attempt to repair the error, I uninstalled HeatWave from VS2022, as well as the WiX V3 libraries to start again. I downloaded HeatWave V4 again (HeatWave.Dev17) and followed the V4 tutorial. At Sprint 1 : Create an MSI package, I get the MS Visual Studio Error Message "Attempted to access a missing method". There is nothing in the project yet, just following the steps outlined in the tutorial. So something has become broken and I don't know what. This misery is post the release of WiX 5.0 so I'm wondering is since that release, V4 has become deprecated?

**Comment by ColinWPA** [comment]
Here is some more info from the Activity Log:

  <entry>
    <record>215</record>
    <time>2024/04/08 13:48:31.718</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>SetSite failed for package [&#x0005;&#x2002;&#x2001;]Source: &apos;FireGiant.VisualStudio&apos; Description: Method not found: &apos;Microsoft.VisualStudio.Threading.JoinableTaskFactory Microsoft.VisualStudio.Shell.ThreadHelper.get_JoinableTaskFactory()&apos;.&#x000D;&#x000A;System.MissingMethodException: Method not found: &apos;Microsoft.VisualStudio.Threading.JoinableTaskFactory Microsoft.VisualStudio.Shell.ThreadHelper.get_JoinableTaskFactory()&apos;.&#x000D;&#x000A;   at FireGiant.VisualStudio.FireGiantPackage.Dispose(Boolean disposing)&#x000D;&#x000A;   at Microsoft.VisualStudio.Shell.Package.Microsoft.VisualStudio.Shell.Interop.IVsPackage.Close()</description>
    <guid>{E20DDE51-8F89-4C4C-B9EA-230A63662CDE}</guid>
    <hr>0x80131513</hr>
    <errorinfo></errorinfo>
  </entry>
![CaptureWiX](https://github.com/wixtoolset/issues/assets/114072950/8b52eca8-f595-4ce5-8d74-d715605b86cd)

**Reply by ColinWPA** [reply]
And some more:

error  : An exception was thrown while initializing part "FireGiant.HeatWave.ProjectSystem.HeatWaveProjectTreeIconPropertyProvider".

**Comment by robmen** [comment, marked as answer]
Sounds like you've hit https://github.com/firegiant/HeatWaveSupport/issues/59. Try upgrading VS2022 to 17.9.

**Reply by ColinWPA** [reply]
Thank you. That solved that issue..... but opened up another, to do with WixUI...

WIX0200	The UI element contains an unhandled extension element 'WixUI'. Please ensure that the extension for elements in the 'http://wixtoolset.org/schemas/v4/wxs/ui' namespace has been provided.

Which is weird as it worked before, but after the upgrade of WixToolset.UI,winext to 5.0.0 the issue shows up. I will scan through the forum to see if this issue has been raised already to see what the solution is. Thanks again for your help - much appreciated

**Reply by ColinWPA** [reply]
Ok, solved it. I needed to update my wixproj file from 4.0.4 to 5.0.0., viz.

Project Sdk="WixToolset.Sdk/5.0.0" ToolsVersion="5.0"

Obviously it was trying to use the 4.0.4 toolset when the toolset has been updated to 5.0.0..
